### PR TITLE
Feature/test python wf

### PR DESCRIPTION
**Pull request by kfang4**
Tested re-written ASAP workflow (R -> Python) with two samples from three teams each. Outputs and file names were inspected and changes have been made to reflect that.

### CODE REVIEW

File: workflows/cohort_analysis/cluster_data/cluster_data.wdl [hunk 1]
```diff
@@ -144,80 +138,39 @@ task cluster_cells {
...
-			#	-i ~{write_tsv(workflow_info)} \
-			#	-o "~{integrated_adata_object_basename}.mde_cluster.h5ad.gz" \
-			#	-o "~{cohort_id}.major_type_module_umap.pdf" \
-			#	-o "~{cohort_id}.major_type_module_umap.png"
-
-		else
-			echo "[ERROR] Choose a valid clustering method; options are 'umap' or 'mde'"
-			exit 1
-		fi
+		# TODO - script doesn't use cell_type_markers_list (incomplete)
```
Comment (hkeward) [inline workflows/cohort_analysis/cluster_data/cluster_data.wdl:149]: Does it now or is this still a TODO?

**Review by hkeward — CHANGES_REQUESTED** [review]
A few small changes, overall looks great!

File: workflows/cohort_analysis/cohort_analysis.wdl [hunk 2]
```diff
@@ -241,49 +232,52 @@ task merge_and_plot_qc_metrics {
...
-			--threads ~{threads} \
-			--adata-objects-fofn adata_objects_paths.txt \
-			--project-name ~{cohort_id} \
-			--adata-output ~{cohort_id}.merged_adata_object.h5ad.gz
-
-		# TODO - double check file name and type for all plots
+			--adata-objects-fofn adata_samples_paths.tsv \
+			--adata-output ~{cohort_id}.merged_adata_object.h5ad
+
+		mv "plots/violin_n_genes_by_counts.png" "plots/~{cohort_id}.n_genes_by_counts.violin.png"
```
[the remark this answers is not in the document]
Comment (kfang4) [reply workflows/cohort_analysis/cohort_analysis.wdl:244]: Unfortunately, scanpy always prepends- for example, with "violin" or "umap" (issue: https://github.com/scverse/scanpy/issues/1508).

File: workflows/main.wdl [hunk 3]
```diff
@@ -189,8 +179,8 @@ workflow harmonized_pmdbs_analysis {
 
 		# Preprocess
 		Array[Array[File]] report_html = preprocess.report_html
-		Array[Array[File]] remove_background_counts = preprocess.remove_background_counts
-		Array[Array[File]] filtered_remove_background_counts = preprocess.filtered_remove_background_counts
+		Array[Array[File]] removed_background_counts = preprocess.removed_background_counts
+		Array[Array[File]] filtered_removed_background_counts = preprocess.filtered_removed_background_counts
 		Array[Array[File]] cell_barcodes_csv = preprocess.cell_barcodes_csv
 		Array[Array[File]] graph_pdf = preprocess.graph_pdf
 		Array[Array[File]] log = preprocess.log
```
[the remark this answers is not in the document]
Comment (kfang4) [reply workflows/main.wdl:186]: I followed the same format as cellranger and prefixed outputs with `cellbender_`, [here](https://github.com/ASAP-CRN/harmonized-wf-dev/blob/c7200305530a671cfbd39c7b6e1c3448407f3554/workflows/preprocess/preprocess.wdl#L51-L59)

[the remark this answers is not in the document]
Comment (hkeward) [reply workflows/main.wdl:186]: I'd prefer if at least in main.wdl the `log` output was called `cellbender_log`. counts files are less ambiguous than `log` but could potentially still do with prefixing. `report_html`, `adata_object`, a bunch of other preprocessing outputs could also hypothetically come from a couple different steps -- I think adding prefixes at least to main.wdl outputs, but also ideally to preprocessing outputs, would help with ambiguity.

I am imagining the eventual workbench interface with outputs, where we'll just get an output called `harmonized_pmdbs_analysis.log` (not the surrounding comments that say it comes from preprocess) -- I don't want whoever is inspecting this to need to refer to the workflow definition to figure out what this `log` refers to.

**Review by hkeward — APPROVED** [review]

**Pull request merged**